### PR TITLE
Milestone4 category filter

### PR DESCRIPTION
**Pull request by truetechcode**
created filter reducer and action
stored filter in redux store

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+function CategoryFilter(props) {
+  const CATEGORIES = ["ALL", "Action", "Biography", "History", "Horror", "Kids", "Learning", "Sci-Fi"]
+
+  return (
+    <select name="category" onChange={(e) => props.onChange(e)}>
+    <option value='default'>Filter Books</option>
```
Comment (SunnyAsar) [inline src/components/CategoryFilter.js]: Hey, this is not needed, i should be seeing `All` its all the books that are displayed at start after all.

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+function CategoryFilter(props) {
+  const CATEGORIES = ["ALL", "Action", "Biography", "History", "Horror", "Kids", "Learning", "Sci-Fi"]
+
+  return (
+    <select name="category" onChange={(e) => props.onChange(e)}>
```
Comment (SunnyAsar) [inline src/components/CategoryFilter.js]: Naming functions the same as event handlers can get tricky maybe `handleChange`[OPTIONAL]

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -1,35 +1,57 @@
...
 
-
 function BooksList(props) {
+
+  const handleFilterChange = (e) => {
+    let filter = e.target.value;
+    props.changeFilter(filter)
+  }
+
+  let filteredBooks = props.books.filter((book) => book.category === props.filter || props.filter === '')
```
Comment (SunnyAsar) [inline src/containers/BooksList.js]: we could do this at the state level and fetch only what we need.

File: src/reducers/filter.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
+const filter = (state = 'ALL', action) => {
+  switch(action.type){
+    case 'ALL':
+      return ''
+    
+    case 'Action':
+      return 'Action'
```
Comment (SunnyAsar) [inline src/reducers/filter.js]: Hi guys we can cut this down significantly like this:

action:
```
export const changeFilter = category => {
  return {
    type: `CHANGE_FILTER`,
    category
  }
}
```

reducer:
this way we can return exactly what is picked in category to be saved in store and use it to filter books.
```
switch (action.type) {
    case `CHANGE_FILTER`:
      return action.filter
    default:
      return state
  }
```

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
# Hey @truetechcode  and @shloch 👋 
### well done guy, project is looking good.
I made some comments below, concerning the option-select like in the previous milestone. also a bout your approach to filter. Please send me slack message if there is any confusion.


**Please request another review with update**

**Review by BalintMendli — APPROVED** [review]
Hi @truetechcode @shloch ,

Great job. It is a
### 👍 

You can merge the project.

**Pull request merged**